### PR TITLE
Expose all tests assigned to user

### PR DESCRIPTION
**Pull request by saulius**
Hello!

So @vinted/data-warehouse is now using this library in the analytics stack. And we need a method to determine all AB test variants for a user - doesn't matter if it is running or has already ended. So I added it in. Not sure about my Java-foo.

@slapsys @mjurkus @ljank

### CODE REVIEW

File: src/test/java/com/vinted/ab/TestConfigurationTest.java [hunk 1]
```diff
@@ -66,6 +66,25 @@ public void testAssignedTests() {
...
+        TestConfiguration input = read(TestConfiguration.class, String.format("/%s/input.json", set));
+        TestOutput output = read(TestOutput.class, String.format("/%s/output.json", set));
+
+        for (Entry<String, JsonElement> element : output.variants.entrySet()) {
+            String variant = element.getKey();
+
+            for (String identifier : output.get(variant)) {
+                List<AbTestVariant> assignedTestVariants = input.getAllTestVariantsForTest(identifier, output.testName);
+
+                for (AbTestVariant test : assignedTestVariants) {
```
Comment (ljank) [inline src/test/java/com/vinted/ab/TestConfigurationTest.java]: assert `assignedTestVariants` is not empty?

File: src/main/java/com/vinted/ab/TestConfiguration.java [hunk 2]
```diff
@@ -67,4 +66,56 @@ public TestConfiguration(String salt, int bucketCount, List<AbTest> abTests) {
...
+    /**
+     * @param identifier user identifier
+     * @return List of all (running + ended) tests assigned to user
+     */
+    public List<AbTest> getAllTests(String identifier) {
+        BigInteger bucketId = getAssignedBucketId(identifier);
+
+        List<AbTest> tests = new ArrayList<>();
+        for (AbTest test : abTests) {
+            if ((test.isAllBuckets() || test.getBuckets().contains(bucketId.intValue()))) {
```
Comment (ljank) [inline src/main/java/com/vinted/ab/TestConfiguration.java]: it's strange that `getAssignedBucketId` returns BigInteger, but the only thing one needs it `intValue()` from it :\ a) why BigInteger at all? b) why not int right away?

Comment (saulius) [inline src/main/java/com/vinted/ab/TestConfiguration.java]: I honestly have no idea, I took this part from `getRunningTests`. @slapsys any ideas? :)

File: src/test/java/com/vinted/ab/TestConfigurationTest.java (hunk 1 above)
Comment (saulius) [inline src/test/java/com/vinted/ab/TestConfigurationTest.java]: :+1:. Added `assertFalse(assignedTestVariants.isEmpty());` before variant checking.

File: src/main/java/com/vinted/ab/TestConfiguration.java (hunk 2 above)
Comment (mjurkus) [inline src/main/java/com/vinted/ab/TestConfiguration.java]: If I remember correctly its because of this part `new BigInteger(hex, 16)`

Comment (saulius) [inline src/main/java/com/vinted/ab/TestConfiguration.java]:
```
Digest::SHA256.hexdigest("6df2c502-425e-481a-bef4-b6d190db6c03").to_i(16)
=> 76673387891688753902299618191560834725956573180094717539529699678503362048745
```

Looks `Big`-ish to me too..

**Comment by saulius** [comment]
So @ljank @mjurkus @slapsys :+1:?

**Comment by mjurkus** [comment]
:+1:

**Comment by ljank** [comment]
just do it :+1:

**Comment by saulius** [comment]
Btw, no changes necessary from Android side - nothing changes there.

File: src/main/java/com/vinted/ab/TestConfiguration.java (hunk 2 above)
Comment (slapsys) [inline src/main/java/com/vinted/ab/TestConfiguration.java]: `getAssignedBucketId` could return `int`.. But it's ok ;)

**Pull request merged**